doc/stream-telemetry/stream-telemetry-hld.md

```diff
@@ -0,0 +1,647 @@
+# Stream telemetry high level design <!-- omit in toc -->
```

Comment:
"Stream telemetry" is a term already used by sonic gnmi based telemetry implementation. Suggest you replace it with another term like "high frequency counters telemetry" (just an example).